src/Computer/Wind/CirclingWind2.cpp

```diff
@@ -152,9 +152,19 @@ CirclingWind2::CalcWind(double quality_metric, const size_t n_samples, [[ maybe_
 
   // reject if average time step greater than 2.0 seconds
   const auto measure_time = (samples[0].time - samples[n_samples-1].time);
-  if ((measure_time / n_samples) > std::chrono::seconds{2})
+  const auto avg_step_width = measure_time / n_samples;
+  if (avg_step_width > std::chrono::seconds{2})
     return Result(0);
 
+  // reject if step width is't sufficiently uniform
+  for (size_t i = 1; i < n_samples; i++)
+    // if (abs(samples[i].time - samples[i-1].time) > std::chrono::milliseconds{100})
+    if ((samples[i].time - samples[i-1].time) > (avg_step_width * 0.05))
```

Comment:
> should be 1.05 instead of 0.05 i believe but you could get away with worse data even. Most likley is that some gps fixes will just be absent rather than 5% late so could atleast check for larger than 2*avg_step_width.
> 
> Should probably also check for maybe smaller than avg_step_width/2

I see the mistake, but I don't understand why it isn't constantly taking that branch. xcsoar.log should be filled with the message "step width not sufficiently uniform", but I don't see it. I seem to misunderstand the arithmetic on the TimeStamp type.

If the requirement is that any step width mustn't be more than 5% off of the average, then the fix would be in my opinion:
`if (abs(samples[i].time - samples[i-1].time - avg_step_width) > (avg_step_width * 0.05))`
But that is probably not how TimeStamp calculates.
Does anybody see whats wrong??

Reply:
Found the issue:
`if (abs(samples[i-1].time - samples[i].time - avg_step_width) > (avg_step_width * 0.07))`
Will push later.